use-cases/bed-allocation/src/main/java/org/acme/bedallocation/domain/Department.java

```diff
@@ -13,19 +15,20 @@ public class Department {
 
     @PlanningId
     private String id;
-
+    private Map<String, Integer> specialismsToPriority;
     private String name;
     private Integer minimumAge = null;
     private Integer maximumAge = null;
-
     private List<Room> rooms;
 
     public Department() {
+        this.specialismsToPriority = new HashMap<>();
     }
 
     public Department(String id, String name) {
         this.id = id;
         this.name = name;
+        this.specialismsToPriority = new HashMap<>();
```

Comment:
Careful about `HashMap`s - if you ever iterate over them, they will introduce non-determinism in your solver. However, if you ever only `get(...)` on them, it's fine.

Reply:
We are not iterating over it. This was a solution to simplify the JSON payload.